### PR TITLE
Unify usage of context in slasher client

### PR DESCRIPTION
**Pull request by 0xKiwi**
**What type of PR is this?**

Improvement

**What does this PR do? Why is it needed?**
This mirrors #5671 to the slasher client, to ensure SIGTERM cancels when it should be, and also makes sure the wrong context isn't passed into functions.

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Build fails

**Comment by shayzluf** [comment]
did you try it in runtime?

**Comment by 0xKiwi** [comment]
> did you try it in runtime?

Yes! Gets cancelled instantly, before it would always hand and I'd have to manually interrupt it.

File: slasher/beaconclient/receivers.go [hunk 1]
```diff
@@ -134,6 +134,10 @@ func (bs *Service) collectReceivedAttestations(ctx context.Context) {
 		case att := <-bs.receivedAttestationsBuffer:
 			atts = append(atts, att)
 		case collectedAtts := <-bs.collectedAttestationsBuffer:
+			if err := ctx.Err(); err != nil {
```
Comment (rauljordan) [inline slasher/beaconclient/receivers.go:137]: If this is in a for select block <-ctx.Err(): should be its own case

File: slasher/detection/listeners.go [hunk 2]
```diff
@@ -56,6 +56,10 @@ func (ds *Service) detectIncomingAttestations(ctx context.Context, ch chan *ethp
 	for {
 		select {
 		case indexedAtt := <-ch:
+			if err := ctx.Err(); err != nil {
```
Comment (rauljordan) [inline slasher/detection/listeners.go:59]: +1 as above

File: slasher/detection/service.go [hunk 3]
```diff
@@ -93,7 +93,7 @@ func (ds *Service) Start() {
 	// our gRPC client to keep detecting slashable offenses.
 	go ds.detectIncomingBlocks(ds.ctx, ds.blocksChan)
 	go ds.detectIncomingAttestations(ds.ctx, ds.attsChan)
-	go ds.detectHistoricalChainData(ds.ctx)
+	//go ds.detectHistoricalChainData(ds.ctx)
```
Comment (rauljordan) [inline slasher/detection/service.go:96]: plz no commented code

**Pull request merged**